scipy/optimize/zeros.py

```diff
@@ -79,27 +78,38 @@ def _results_select(full_output, r):
     return x
 
 
-# Newton-Raphson method
 def newton(func, x0, fprime=None, args=(), tol=1.48e-8, maxiter=50,
-           fprime2=None,
-           full_output=False, disp=True):
+           fprime2=None, full_output=False, disp=True, failure_idx_flag=None):
     """
     Find a zero using the Newton-Raphson or secant method.
 
     Find a zero of the function `func` given a nearby starting point `x0`.
     The Newton-Raphson method is used if the derivative `fprime` of `func`
     is provided, otherwise the secant method is used.  If the second order
-    derivative `fprime2` of `func` is provided, then Halley's method is used.
+    derivative `fprime2` of `func` is also provided, then Halley's method is
+    used.
+
+    If `x0` is a sequence, then `newton` returns an array, and `func` must be
+    vectorized and return a sequence or array of the same shape as it's first
+    argument. If an optional argument, `failure_idx_flag`, is `True`, then the
+    return is a named tuple `(root, failures, zero_der)` in which `root` is an
+    array of the locations where `func` is zero, `failures` is an array, same
+    size as `root`, of booleans that are `True` where `func` failed to converge,
+    and `zero_der` is another boolean array of the same size where `func` had a
+    zero derivative.
+
+    If `x0` is a sequence, then arguments `full_output` and `disp` are ignored.
```

Comment:
When running `_array_newton`, it makes sense to ignore `disp`.  Implementing it would cause many messages to need to be displayed, clogging up the UI, and there is an alternate mechanism to find out what didn't converge.

`full_output` causes scalar newton to return a RootResults object of `root, converged, iterations, function_calls, error flag`.  Returning one of those RootResults for each `x0` element may not be so useful: `converged` is already obtainable via `failure_idx_flag`, and the `iterations`, `function_calls` would be the same for every `x` (unless tracking separately for each `x0`, which would involve more storage and compute.) 
Do you foresee a need to know the counts of `function_calls` or `iterations` from the vectorized version?

Reply:
I think maybe yes, it might be useful to have the iteration count for each element in the array, but right now, I'm really hoping not to make any more changes unless they're blockers.